templates/redis/bin/common.sh

```
@@ -32,13 +55,22 @@ function log_error() {
+    # Create the target directory structure upfront
+    mkdir -p /var/lib/config-data/generated/var/lib/redis
+    if [ ! -d /var/lib/config-data/generated/var/lib/redis ]; then
+        log_error "Failed to create /var/lib/config-data/generated/var/lib/redis"
+        exit 1
+    fi
+
+    # Copying config files except template files (including symlinks)
+    (cd /var/lib/config-data/default && find . \( -type f -o -type l \) ! -name '.*' ! -name '*.in' -exec sh -c 'mkdir -p "/var/lib/config-data/generated/$(dirname "$1")" && cp -fL "$1" "/var/lib/config-data/generated/$1"' _ {} \;)
+
     # Generating config files from templates
     cd /var/lib/config-data/default
```

Shellcheck (0.11.0)

[warning] Use 'cd ... || exit' or 'cd ... || return' in case cd fails.

(SC2164)